service/endpoint_validation.go

```diff
@@ -97,8 +103,72 @@ func ValidateEndpointCreateRequest(dao dao.EndpointDao, ecr *model.EndpointCreat
 		return fmt.Errorf("the certificate authority cannot be empty")
 	}
 
-	if !util.SliceContainsString(validAvailabilityStatuses, ecr.AvailabilityStatus) {
-		return fmt.Errorf("invalid availability status")
+	// The team decided that the availability statuses will default to "in_progress" whenever they come empty, since
+	// setting them as "unavailable" by default may lead to some confusion to the calling clients.
+	if ecr.AvailabilityStatus == "" {
+		ecr.AvailabilityStatus = model.InProgress
```

Comment:
I see we don't have InProgress in the validAvailabilityStatuses map above - I also think we should just have one instance of that if we could (probably in the model package).

Reply:
Yes, I checked the OpenApi spec and I saw that we only allow the "available" and "unavailable" statuses for the endpoint. That's why I borrowed the "in_progress" one from the model package, and created a custom map for the endpoint's valid statuses.

Do I simply merge "all the structs" into one and let the endpoints have those two values too?

Reply:
I guess I more meant that we have this map here in this file: https://github.com/MikelAlejoBR/sources-api-go/blob/acedd66999d31f744f5475dc3c8186318ea2869f/service/endpoint_validation.go#L26-L31

And also this map over in model: https://github.com/MikelAlejoBR/sources-api-go/blob/acedd66999d31f744f5475dc3c8186318ea2869f/model/availability_status.go#L20-L25

Could we just change this line below to pull from the model package rather than duplicate the map? 
```suggestion
diff --git a/service/endpoint_validation.go b/service/endpoint_validation.go
index 5afa823..141c125 100644
--- a/service/endpoint_validation.go
+++ b/service/endpoint_validation.go
@@ -107,7 +107,7 @@ func ValidateEndpointCreateRequest(dao dao.EndpointDao, ecr *model.EndpointCreat
 	if ecr.AvailabilityStatus == "" {
 		ecr.AvailabilityStatus = model.InProgress
 	} else {
-		if _, ok := validAvailabilityStatuses[ecr.AvailabilityStatus]; !ok {
+		if _, ok := model.ValidEndpointAvailabilityStatuses[ecr.AvailabilityStatus]; !ok {
 			return fmt.Errorf("invalid availability status")
 		}
 	}
```

Reply:
Oops, I thought one of them had the 4 availability statuses... I pushed the change!